src/prefect/utilities/callables.py

```diff
@@ -47,6 +47,82 @@ def get_call_parameters(
     return dict(bound_signature.arguments)
 
 
+def explode_variadic_parameter(
```

Comment:
Simple dedicated tests for these utilities should probably be added before merging.

Reply:
You basically have doctests written there already, but I guess you guys don't have doctests turned on in your pytest.ini?..
I highly recommend doctests. :)
https://docs.pytest.org/en/7.1.x/how-to/doctest.html

Basically just replace your triple-backtick blocks with `>>>`. That is, change this:
```python
    Example:

        ```python
        def foo(a, b, **kwargs):
            pass

        parameters = {"a": 1, "b": 2, "kwargs": {"c": 3, "d": 4}}
        explode_variadic_parameter(foo, parameters)
        # {"a": 1, "b": 2, "c": 3, "d": 4}
        ```
```

...into this:
```python
    Example:

        >>> def foo(a, b, **kwargs):
        ...     pass

        >>> parameters = {"a": 1, "b": 2, "kwargs": {"c": 3, "d": 4}}
        >>> explode_variadic_parameter(foo, parameters)
        {'a': 1, 'b': 2, 'c': 3, 'd': 4}
```

Reply:
I used to write my examples like that but I don't our docs renderer can handle those so we've been switching. It'd be great to enable doctests though if we can get that sorted out :)

Reply:
> our docs renderer can handle those

Awwww. :)

It seems to work fine for me... I cloned the prefect repo, did `pip install -e ".[dev]"` as instructed by the [Contributing page](https://docs.prefect.io/contributing/overview/), and then ran `mkdocs serve`.

I added both styles of Python snippet to this docstring:
```python
class EnforceMinimumAPIVersion:
    """
    FastAPI Dependency used to check compatibility between the version of the api
    and a given request.

    Looks for the header 'X-PREFECT-API-VERSION' in the request and compares it
    to the api's version. Rejects requests that are lower than the minimum version.

    Examples:

        >>> x, y = 1, 2
        >>> z = x + y
        >>> x, y, z
        (1, 2, 3)

        ```python
        x, y = 1, 2
        z = x + y
        x, y, z
        # (1, 2, 3)
        ```

    """
```

...and the resulting page at http://localhost:8000/api-ref/orion/api/dependencies/ looks like this:
![Screenshot from 2023-01-17 19-23-35](https://user-images.githubusercontent.com/101958884/213076078-c0f3b873-46a8-4fd2-83a0-c8b191a19c73.png)

...so maybe there's something I don't know about, but AFAICT, doctests and mkdocs should work just fine together!

Reply:
hm interesting! I'd be happy to go back to that :) thanks for checking! cc @tpdorsey

Reply:
These kinds of examples in the docstrings will always render fine _in the source code view_. There is a small hitch with the new mkdocstrings Python handler and griffe -- it will render certain examples as an "Example" callout box. In that view, `>>>` renders as nested quoted text. But there's a subtlety about which examples get pulled out that I need to look into.

@madkinsz if you see value in pursuing doctests, I'll spend some cycles looking how we can differentiate between the examples used for doctests and examples we want to display in callouts.

Reply:
It'd be good to have test coverage of our examples, but I think we probably have some higher priority docs tooling work right now.